source/image/ImageSet.cpp

```diff
@@ -151,34 +146,30 @@ namespace {
 
 
 // Check if the given path is to an image of a valid file type.
-bool ImageSet::IsImage(const string &path)
+bool ImageSet::IsImage(const filesystem::path &path)
 {
-	if(path.length() < 4)
-		return false;
-
-	string ext = path.substr(path.length() - 4);
+	filesystem::path ext = path.extension();
```

Comment:
It's a path. https://en.cppreference.com/w/cpp/filesystem/path/extension

It's partly because the string's encoding depends on the host system, so it's abstracted away as a path. ~~It's also why Windows is so much trouble~~